# HG changeset patch
# User Kacper Słomiński <kacper.slominski72@gmail.com>
# Date 1575072786 -3600
#      Sat Nov 30 01:13:06 2019 +0100
# Node ID 39a28a4033f5e22b6d3d98ca18c520443b326345
# Parent  bc90ce38f1e27ace54b83bebf987993002504f7f
Add managarm-specific things to build system

diff -r bc90ce38f1e2 -r 39a28a4033f5 CMakeLists.txt
--- a/CMakeLists.txt	Tue Jul 23 14:41:00 2019 -0500
+++ b/CMakeLists.txt	Sat Nov 30 01:13:06 2019 +0100
@@ -112,6 +112,8 @@
     set(AIX TRUE)
   elseif(CMAKE_SYSTEM_NAME MATCHES "Minix.*")
     set(MINIX TRUE)
+  elseif(CMAKE_SYSTEM_NAME MATCHES "managarm")
+    set(MANAGARM TRUE)
   endif()
 elseif(APPLE)
   if(CMAKE_SYSTEM_NAME MATCHES ".*Darwin.*")
@@ -1117,6 +1119,26 @@
     check_include_file("fcitx/frontend.h" HAVE_FCITX_FRONTEND_H)
   endif()
 
+  if(MANAGARM)
+    check_c_source_compiles("
+        #include <linux/input.h>
+        #ifndef EVIOCGNAME
+        #error EVIOCGNAME() ioctl not available
+        #endif
+        int main(int argc, char** argv) {}" HAVE_INPUT_EVENTS)
+
+    set(CORE_LINUX_SOURCES ${SDL2_SOURCE_DIR}/src/core/linux/SDL_evdev.c ${SDL2_SOURCE_DIR}/src/core/linux/SDL_evdev_kbd.c ${SDL2_SOURCE_DIR}/src/core/linux/SDL_udev.c)
+    set(SOURCE_FILES ${SOURCE_FILES} ${CORE_LINUX_SOURCES})
+
+    set(EXTRA_LIBS udev ${EXTRA_LIBS})
+
+    if(HAVE_INPUT_EVENTS)
+      set(SDL_INPUT_LINUXEV 1)
+    endif()
+
+    check_include_file("libudev.h" HAVE_LIBUDEV_H)
+  endif()
+
   if(INPUT_TSLIB)
     check_c_source_compiles("
         #include \"tslib.h\"
diff -r bc90ce38f1e2 -r 39a28a4033f5 cmake/sdlchecks.cmake
--- a/cmake/sdlchecks.cmake	Tue Jul 23 14:41:00 2019 -0500
+++ b/cmake/sdlchecks.cmake	Sat Nov 30 01:13:06 2019 +0100
@@ -760,11 +760,8 @@
 # - nada
 macro(CheckOpenGLESX11)
   if(VIDEO_OPENGLES)
-    check_c_source_compiles("
-        #define EGL_API_FB
-        #include <EGL/egl.h>
-        int main (int argc, char** argv) {}" HAVE_VIDEO_OPENGL_EGL)
-    if(HAVE_VIDEO_OPENGL_EGL)
+    pkg_check_modules(GLES_EGL REQUIRED egl)
+    if(GLES_EGL_FOUND)
         set(SDL_VIDEO_OPENGL_EGL 1)
     endif()
     check_c_source_compiles("
